commands/api.go

```diff
@@ -768,6 +831,19 @@ func parseApi(cmd *cobra.Command, args []string) (*whisk.Api, *QualifiedName, er
 		api.GatewayRelPath = args[0] // Maintain case as URLs may be case-sensitive
 	}
 
+	// Attempting to use path parameters, lets validate that they provided them correctly.
+	hasPathParams := hasPathParameters(api.GatewayRelPath)
+	if err = validatePathParameters(api.GatewayRelPath); hasPathParams && err != nil {
```

Comment:
Seems like you can get around calling the same regex match twice here by making `hasPathParameters` a multi-value retruning function, which returns the parameter and the error.